src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1123,8 +1123,22 @@ is_create_from_half_vectors_overload (MonoMethodSignature *fsig)
 static gboolean
 is_element_type_primitive (MonoType *vector_type)
 {
-	MonoType *element_type = get_vector_t_elem_type (vector_type);
-	return MONO_TYPE_IS_VECTOR_PRIMITIVE (element_type);
+	if (vector_type->type == MONO_TYPE_GENERICINST)
+	{
+		MonoType *element_type = get_vector_t_elem_type (vector_type);
+		return MONO_TYPE_IS_VECTOR_PRIMITIVE (element_type);
+	}
+	else
+	{
+		MonoClass *klass = mono_class_from_mono_type_internal (vector_type);
+		g_assert (
```

Comment:
I am not sure if we want to assert here, since mono won't disable assertion in release build. I think it is better to return false when the class is not one of those expected ones.

Reply:
👍, didn't realize Mono asserts were preserved in release code.